contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -118,6 +129,53 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     emit WatchlistUpdated();
   }
 
+  /// @notice Adds a new address to the watchlist
+  /// @param targetAddress the address to be added to the watchlist
+  /// @param dstChainSelector carries a non-zero value in case the targetAddress is an onRamp, otherwise it carries a 0
+  /// @dev this function has to be compatible with the event onRampSet(address, dstChainSelector) emitted by
+  /// the CCIP router. Important detail to know is this event is also emitted when an onRamp is decomissioned,
+  /// in which case it will carry the proper dstChainSelector along with the 0x0 address
+  function addToWatchListOrDecomission(
+    address targetAddress,
+    uint64 dstChainSelector
+  ) public onlyRoleOrAdminRole(EXECUTOR_ROLE) {
+    if (s_targets[targetAddress].isActive) revert DuplicateAddress(targetAddress);
+    address oldAddress = s_onRampAddresses[dstChainSelector];
+    // if targetAddress is an existing onRamp, there's a need of cleaning the previous onRamp associated to this dstChainSelector
+    // there's no need to remove any other address that's not an onRamp
+    if (dstChainSelector > 0 && oldAddress != address(0)) {
+      removeFromWatchList(oldAddress);
+    }
+    // only add the new address if it's not 0x0
+    if (targetAddress != address(0)) {
+      s_onRampAddresses[dstChainSelector] = targetAddress;
+      s_targets[targetAddress] = MonitoredAddress({
+        isActive: true,
+        minBalance: DEFAULT_MIN_BALANCE_JULES,
+        topUpAmount: DEFAULT_TOP_UP_AMOUNT_JULES,
+        lastTopUpTimestamp: 0
+      });
+      s_watchList.push(targetAddress);
+    } else {
+      // if the address is 0x0, it means the onRamp has ben decomissioned and has to be cleaned
+      delete s_onRampAddresses[dstChainSelector];
+    }
+  }
+
+  /// @notice Delete an address from the watchlist and sets the target to inactive
+  /// @param targetAddress the address to be deleted
+  function removeFromWatchList(address targetAddress) public onlyRoleOrAdminRole(EXECUTOR_ROLE) returns (bool) {
+    s_targets[targetAddress].isActive = false;
+    for (uint256 i; i < s_watchList.length; i++) {
+      if (s_watchList[i] == targetAddress) {
+        s_watchList[i] = s_watchList[s_watchList.length - 1];
+        s_watchList.pop();
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
this is going to be rather expensive... do we know how large we expect `s_watchlist` to be? If it's going to be larger than, say 10... we should probably find another solution.

Reply:
I don't think we can have a right estimate on the watchlist length, but this function will be called very sporadically (if at all) and it's here only to ensure CL Automation adds onRamps correctly.